src/features/os-config/index.ts

```diff
@@ -1,27 +1,45 @@
-import type { Application, RequestHandler } from 'express';
+import type { Application } from 'express';
 
 import {
 	DEVICE_CONFIG_OPENVPN_CA,
 	DEVICE_CONFIG_OPENVPN_CONFIG,
 	DEVICE_CONFIG_SSH_AUTHORIZED_KEYS,
+	LOGS_HOST,
 } from '../../lib/config';
 import { b64decode } from '../../lib/utils';
 
-const getOsConfiguration: RequestHandler = (_req, res) => {
-	res.json({
-		services: {
-			openvpn: {
-				config: DEVICE_CONFIG_OPENVPN_CONFIG,
-				ca: b64decode(DEVICE_CONFIG_OPENVPN_CA),
-			},
-			ssh: {
-				authorized_keys: DEVICE_CONFIG_SSH_AUTHORIZED_KEYS,
-			},
-		},
-		schema_version: '1.0.0',
-	});
+// OS service configurations
+const services = {
+	openvpn: {
+		config: DEVICE_CONFIG_OPENVPN_CONFIG,
+		ca: b64decode(DEVICE_CONFIG_OPENVPN_CA),
+	},
+	ssh: {
+		authorized_keys: DEVICE_CONFIG_SSH_AUTHORIZED_KEYS,
+	},
+};
+
+// Config.json migrations: changes should be evaluated for security risks before applying.
+const config = {
+	// config.json fields to add or change
+	update: {
+		logsEndpoint: LOGS_HOST != null ? `https://${LOGS_HOST}` : undefined,
+	},
+	// config.json fields to delete
+	delete: [],
 };
 
 export const setup = (app: Application) => {
-	app.get('/os/v1/config/', getOsConfiguration);
+	app.get('/os/v1/config/', (_req, res) => {
+		res.json({
+			services,
+			schema_version: '1.0.0',
+		});
+	});
+	app.get('/os/v2/config', (_req, res) => {
+		res.json({
+			services,
+			config,
```

Comment:
Is it fine to not have `schema_version` defined here as well?

Reply:
After discussing the API changes in an architecture sync, we agreed to keep the legacy schema_version field for now and not introduce a new /os/v2/config endpoint